src/Workspaces/Core/Portable/Workspace/Solution/Solution.cs

```diff
@@ -665,13 +636,15 @@ public Solution RemoveProjectReference(ProjectId projectId, ProjectReference pro
 
             CheckContainsProject(projectId);
 
-            var newState = _state.RemoveProjectReference(projectId, projectReference);
+            // If the project didn't change itself, there's no need to change the compilation state.
+            var (newState, newProjectState) = _state.RemoveProjectReference(projectId, projectReference);
             if (newState == _state)
             {
                 throw new ArgumentException(WorkspacesResources.Project_does_not_contain_specified_reference, nameof(projectReference));
             }
```

Comment:
note: this pattern is slightly different, because we do some argument checking within Solution.  i liked Solution being the only thing responsible for validation.  Once we call into SolutionState or SOlutionCompilationState it just always does the work.

Reply:
Should this just check the project reference existing more direclty rather than having the different pattern here?